ncar12km_decadal_summaries/compute_summaries.py

```diff
@@ -0,0 +1,208 @@
+    
+    # Down left corner of the domain
+    x0 = -(nx-1) / 2. * dx + e
+    y0 = -(ny-1) / 2. * dy + n
+    # 2d grid
+    x = np.arange(nx) * dx + x0
+    y = np.arange(ny) * dy + y0
+    
+    projected_datacube = datacube.assign_coords({"y": ("y", y), "x": ("x", x)})
+    
+    # Output geotiff creation profile params  
+    width = datacube.x.shape[0]
+    height = datacube.y.shape[0]
+    
+    # west and north
+    west = x0 - dx / 2
+    north = y[-1] + dy / 2
+    out_transform = rio.transform.from_origin(west, north, dx, dy)
+    
+    wrf_raster_profile = {
+        "driver": "GTiff",
+        "crs": CRS.from_proj4(wrf_proj_str),
+        "transform": out_transform,
+        "width": width,
+        "height": height,
+        "count": 1,
+        "dtype": np.float32,
+        "nodata": -9999,
+        "tiled": False,
+        "compress": "lzw",
+        "interleave": "band",
+    }
+    
+    datacube.close()
+    
+    return projected_datacube, wrf_raster_profile
+
+
+def list_years_in_decade(start_year):
+    """
+    Given a starting year, return a list of ten consecutive years representing a decade.
+
+    Args:
+        start_year (int): starting year of the decade
+
+    Returns:
+        list: A list of ten consecutive years
+    """
+    return list(range(start_year, start_year + 10))
+
+
+def slice_by_decade(datacube, decade_start_year):
+    """
+    Given a xarray datacube and a starting year, slice the datacube to select only
+    the data from the decade corresponding to the starting year.
+
+    Args:
+        datacube (xarray.Dataset): multidimensional datacube
+        decade_start_year (int): starting year of the decade to be sliced
+
+    Returns:
+        decade_slice (xarray.Dataset): A sliced xarray datacube containing only the data from the
+        specified decade (e.g. 2030 through 2039)
+    """
+    years = list_years_in_decade(decade_start_year)
+    decade_slice = datacube.isel(time=datacube.time.dt.year.isin(years))
+    return decade_slice
+
+
+def compute_monthly_summaries(decade_slice, vargroup, climvar):
+    """
+    Compute monthly summaries like mean, total, etc. of a climatological variable over a decadal slice of data.
+    
+    Args:
+        decade_slice (xarray.Dataset): a slice of data corresponding to a decade.
+        vargroup (str): category of variables to summarize (one of wf, ws, eb, or met)
+        climvar (str): name of the climatological variable to summarize.
+    
+    Returns:
+        dec_mean_monthly_summary (xr.DataArray): a dataset containing the decadal monthly summaries of the given climatological variable.
+    """
+    
+    summary_func = variable_di[vargroup][climvar]
+    out = (
+        decade_slice[climvar]
+        .resample(time="1M")
+        .reduce(summary_func)
+        .groupby("time.month")
+        .reduce(np.mean)  # decadal summary is always a mean
+    )
+    dec_mean_monthly_summary = out.compute()
```

Comment:
Charlie, you call ```compute``` here but haven't imported ```dask``` in this script.  Is this an ```xarray``` function in this case? Or are you setting this up here because you know you will use ```dask``` in the notebooks?